### DISCUSSION TITLE
[Help needed] Conflict with TextBox and HotKeys in MenuItem (Copy, Paste, Cut)

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#10902] [Q&A]
Hi all,

I added Cut, Copy, Paste menu items with hotkeys and commands.
But when a textbox is focused, and that I use hotkey such as Ctrl+V, it is the command in the menu item which is called, and nothing changed in the textbox.
Is there a way to make the focused control to handle the hotkeys first, and the menu items only if textbox has not handled it?

Another question about hotkeys, currently the menu item need to be show at least once in order to make the hotkey work. Is there a alternate way to deal with hotkeys in order to make it work before the menu items was displayed? a manager or something.

Bonus question: is there a way to make Copy/Paste in textboxes work without displaying the prompt for permission in browser? I suppose not, but may be there is a magical trick :)

Thank you.

**Comment by dbriard** [comment]
1) Finally, I stopped using controls builtin HotKey property which and handled my hokeys manually in TopLevel's KeyDown event, if e.Source is not a TextBox, using similar method to what Avalonia TextBox KeyDown is doing.

`toplevel.AddHandler(InputElement.KeyDownEvent, HandleKeyDown, handledEventsToo: true);`

If anyone got a cleaner way to do that, I am open.

2) Also, does anyone know if there is a way to prevent the browser's authorize prompt when we paste into a TextBox? By inspecting with the browser developer tools, it look like there is a floating html text area that position above the TextBox, and I am thinking it is to prevent the prompt, but if it is that, it do not work.

**More blocking**, it seems that with Firefox, there is no prompt, and we cannot paste at all. Can we do something about that?

**Comment by mpulukkinen** [comment]
I have the same issue, so gonna do just like you then
